Sources/Sentry/SentrySession.m

```diff
@@ -9,61 +9,92 @@ @implementation SentrySession
     }
     return self;
 }
 
 - (instancetype)initWithJSONObject:(NSDictionary *)jsonObject
 {
-    // Note this doesn't use the main init method since it should only init
-    // fields that exist in the JSON.
-    if (self = [super init]) {
-        _sessionId = [[NSUUID UUID] initWithUUIDString:[jsonObject valueForKey:@"sid"]];
-        _distinctId = [jsonObject valueForKey:@"did"];
-        NSString *startedString = [jsonObject valueForKey:@"started"];
-        if (nil != startedString) {
-            _started = [NSDate sentry_fromIso8601String:startedString];
-        }
-        NSString *timestampString = [jsonObject valueForKey:@"timestamp"];
-        if (nil != timestampString) {
-            _timestamp = [NSDate sentry_fromIso8601String:timestampString];
-        }
-        NSString *status = [jsonObject valueForKey:@"status"];
-        if ([@"ok" isEqualToString:status]) {
-            _status = kSentrySessionStatusOk;
-        } else if ([@"exited" isEqualToString:status]) {
-            _status = kSentrySessionStatusExited;
-        } else if ([@"crashed" isEqualToString:status]) {
-            _status = kSentrySessionStatusCrashed;
-        } else if ([@"abnormal" isEqualToString:status]) {
-            _status = kSentrySessionStatusAbnormal;
-        }
-        _sequence = [[jsonObject valueForKey:@"seq"] unsignedIntegerValue];
-        _errors = [[jsonObject valueForKey:@"errors"] unsignedIntegerValue];
+    // We use the default constructor here to set the non nullable values to a default values,
+    // because this could cause crashes, for example, in serialize.
+    // With this approach we avoid crashes and accept the tradeoff that some session data might not
+    // be 100% accurate.
+    if (self = [self init]) {
+        NSUUID *sessionId = [[NSUUID UUID] initWithUUIDString:[jsonObject valueForKey:@"sid"]];
+        if (nil != sessionId) {
+            _sessionId = [[NSUUID UUID] initWithUUIDString:[jsonObject valueForKey:@"sid"]];
+        }
+
+        if ([[jsonObject valueForKey:@"started"] isKindOfClass:[NSString class]]) {
+            _started = [NSDate sentry_fromIso8601String:[jsonObject valueForKey:@"started"]];
+        }
+
+        if ([[jsonObject valueForKey:@"status"] isKindOfClass:[NSString class]]) {
+            NSString *status = [jsonObject valueForKey:@"status"];
+            if ([@"ok" isEqualToString:status]) {
+                _status = kSentrySessionStatusOk;
+            } else if ([@"exited" isEqualToString:status]) {
+                _status = kSentrySessionStatusExited;
+            } else if ([@"crashed" isEqualToString:status]) {
+                _status = kSentrySessionStatusCrashed;
+            } else if ([@"abnormal" isEqualToString:status]) {
+                _status = kSentrySessionStatusAbnormal;
+            }
+        }
+
+        if (nil != [jsonObject valueForKey:@"seq"]) {
+            _sequence = [[jsonObject valueForKey:@"seq"] unsignedIntegerValue];
```

Comment:
H: Two map lookups aren't needed here. You can stash the value to avoid a second lookup